hydra-node/src/Hydra/Options.hs

```diff
@@ -133,28 +133,14 @@ commandParser =
           (progDesc "Generate a pair of Hydra signing/verification keys (off-chain keys).")
       )
 
-data PublishOptions = PublishOptions
-  { publishNetworkId :: NetworkId
-  , publishNodeSocket :: SocketPath
-  , publishSigningKey :: FilePath
+newtype PublishOptions = PublishOptions
+  { publishChainConfig :: ChainConfig
   }
   deriving stock (Show, Eq)
 
--- | Default options as they should also be provided by 'runOptionsParser'.
-defaultPublishOptions :: PublishOptions
-defaultPublishOptions =
-  PublishOptions
-    { publishNetworkId = Testnet (NetworkMagic 42)
-    , publishNodeSocket = "node.socket"
-    , publishSigningKey = "cardano.sk"
-    }
-
 publishOptionsParser :: Parser PublishOptions
 publishOptionsParser =
-  PublishOptions
-    <$> networkIdParser
-    <*> nodeSocketParser
-    <*> cardanoSigningKeyFileParser
+  PublishOptions <$> chainConfigParser
```

Comment:
Must: not take more arguments than we need

This one is requesting `--hydra-scripts-tx-id`, `--hydra-verification-key`, etc. Also, it will allow providing `--offline-head-seed` on the publish subcommand which does not make sense at all